orttraining/orttraining/core/optimizer/transformer_layer_recompute.cc

```diff
@@ -9,42 +9,82 @@
 #include <deque>
 
 namespace onnxruntime {
+
+namespace {
+bool IsOpType(const Node& node, const std::unordered_set<std::string>& op_types) {
+  return op_types.count(node.OpType()) > 0;
+}
+
+}  // namespace
+
 Status TransformerLayerRecompute::IdentifyTransformerLayerEdges(
     const Graph& graph,
-    std::vector<std::pair<const NodeArg*, const NodeArg*>>& start_end_edges,
+    std::vector<const NodeArg*>& layer_start_edges,
+    std::vector<const NodeArg*>& layer_end_edges,
     const logging::Logger& logger) const {
   const std::unordered_set<std::string> gelu_ops{"Gelu", "BiasGelu", "FastGelu"};
   const std::unordered_set<std::string> dropout_ops{"Dropout", "BiasDropout"};
-  const std::unordered_set<std::string> layernorm_ops{"LayerNormalization", "SkipLayerNormalization"};
+  const std::unordered_set<std::string> layernorm_ops{"LayerNormalization", "SkipLayerNormalization", "SimplifiedLayerNormalization"};
+  const std::unordered_set<std::string> matmul_ops{"MatMul", "FusedMatMul", "TransposeMatMul"};
+  const std::unordered_set<std::string> transpose_ops{"Transpose"};
+
+  layer_start_edges.clear();
+  layer_end_edges.clear();
 
-  std::vector<const NodeArg*> layer_start_edges, layer_end_edges;
   GraphViewer graph_viewer(graph);
   const auto& node_topology_list = graph_viewer.GetNodesInTopologicalOrder();
   for (auto node_index : node_topology_list) {
     auto& node = *graph.GetNode(node_index);
 
-    // Look for start of a transformer layer
-    if ((layernorm_ops.find(node.OpType()) != layernorm_ops.end() ||
-         dropout_ops.find(node.OpType()) != dropout_ops.end()) &&
+    /*
+    Look for start of a transformer layer
+          [LayerNorm|Dropout|Transpose]
```

Comment:
seems we won't handle the case when we enable megatron, right? then we will have a MegatronF after layernorm, before QKV usually.